server/src/item.rs

```diff
@@ -28,8 +28,39 @@ pub struct Item {
 #[zbus::interface(name = "org.freedesktop.Secret.Item")]
 impl Item {
     #[zbus(out_args("prompt"))]
-    pub async fn delete(&self) -> Result<ObjectPath, ServiceError> {
-        todo!()
+    pub async fn delete(
+        &self,
+        #[zbus(object_server)] object_server: &zbus::ObjectServer,
+    ) -> Result<OwnedObjectPath, ServiceError> {
+        let manager = self.manager.lock().await;
+
+        let Some(collection) = manager.collection(&self.collection_path) else {
+            return Err(ServiceError::NoSuchObject(format!(
+                "Collection `{}` does not exist.",
+                &self.collection_path
+            )));
+        };
+
+        drop(manager);
+        match collection.delete_item(&self.path).await {
+            Ok(_) => {
+                tracing::info!("Item `{}` deleted.", &self.path);
+            }
+            Err(_) => {
+                return Err(ServiceError::NoSuchObject(format!(
```

Comment:
Is this the only possible error? I don't think so. It would be nice if delete_item returned a ServiceError already as it is only used here and then you can just call collection.delete_item(&self.path).await?;

Reply:
Done.